### ISSUE TITLE
Incorrect directions, turn right and turn left in zero feet

### ISSUE THREAD

**Issue by seahorse41** [osmandapp/Osmand#6043]
Driving along a straight highway at 100 kph, I hear directions to turn, when the arrow on screen shows keep straight. I've transcribed the spoken directions from dashcam video.

Version 3.1.6 for Android on Samsung G900A
Offline map Canada_british-columbia_northamerica_2.obf.zip 2018-09-01
routing from home, through Grand Forks to Castlegar (by Highway 3)
to a specific destination past Castlegar, and includes some intermediate destinations along the way, the closest being a rest area north of Christina Lake behind me and no longer in the list, and a turnout ahead entering Castlegar.
routing mode is car
navigation service = OSMAND
Fastest route = No
Allow private access = Yes

"In 4/10 of a mile turn slightly right onto Paulson Cross Country Recreation Site, and keep left."
"Turn slightly right onto Paulson Cross Country Recreation Site, then in zero feet turn slightly left."

The intersection in question is at 49.24426 -118.00146

I occasionally see these zero feet errors. So upon getting home I loaded OSM editor to "fix" this. But the nodes look correct, attached, and no duplicates. I did adjust the angle of the intersection to the mentioned side road, so future obf map releases, or online maps, might act different. 
In short I see no excuse for this, looks like a genuine bug.
Can somebody duplicate and determine the cause?
Thanks

**Comment by seahorse41** [comment]
I traveled the same route again, but without any Intermediate destinations. This time it choose a different place to show the bug. So the route I take is a variable affecting this behavior. Maybe even stopping at a rest area, triggering a "route recalculated" message could be a factor in when this shows up.
This time I recorded it as a video.
From where I turned back onto the highway the OSM voice said:
"Continue for about 8 miles to Paulson Detour Road."
then 2 minutes later very unusual to hear repeated messages Between waypoints:
"Continue for about 6 miles to Paulson Detour Road."
Then 5 minutes later the usual and expected normal place to hear directions:
"After 9 tenths of a mile, turn slightly left on Paulson Detour Road."
and again at 3/10 of a mile
Then just before the corner in question:
"Turn slightly left on Paulson Detour Road."
Passing the intersection with a one-lane gravel trail at 80 KPH because I'm not following those directions,
"Continue for 31 miles."
Where can I send or post video or screenshots from that video?

**Comment by vshcherb** [comment]
Just tested with latest data, it shows turn right. So it might be was a mistake in map data which was already corrected.
